lua/neogit/buffers/commit_editor/init.lua

```diff
@@ -40,24 +40,24 @@ function M:open()
     modifiable = true,
     readonly = false,
     autocmds = {
-      ["BufWritePost"] = function()
-        written = true
```

Comment:
why did you remove the written condition?

Reply:
As far as I could see, we could achieve the desired functionality with a negating guard clause condition. Which is less nested with less code. Also, I've seen the crook in here that this is the reason why commits are always send. As it would always be true due to the hard-coded save commands. 

And the rebase editor also has none: https://github.com/TimUntersberger/neogit/blob/3323380e4a110ec3bfa4d2a681f4a9634a239e87/lua/neogit/buffers/rebase_editor/init.lua#L27-L30 so I'll burned it.

Alternatively it should work to use the original with less modification and additionally add an else statement to return the function early.